src/p8est_communication.h

```diff
@@ -564,6 +564,181 @@ void                p8est_transfer_items_end (p8est_transfer_context_t * tc);
+ * \ref p8est_transfer_search_gfx and \ref p8est_transfer_search_gfp.
+ * 
+ * \param[in] p8est In the versions of transfer search not requiring an
+ *                  explicit p8est this is a dummy p8est where only the user
+ *                  pointer is initialized.
+ * \param[in] which_tree Tree containing quadrant
+ * \param[in] quadrant The quadrant
+ * \param[in] point The point
+ * \return True if \a point intersects \a quadrant.
+*/
+typedef int         (*p8est_intersect_t) (p8est_t *p8est,
+                                          p4est_topidx_t which_tree,
+                                          p8est_quadrant_t * quadrant,
+                                          void *point);
+
+/** This structure is used with \ref p8est_transfer_search to maintain a
+ * distributed collection of points, so that the points known to a process
+ * are exactly the points which intersect its domain. Points are completely
+ * arbitrary and may for example represent geometric objects such as polyhedra
+ * or geodesics.
+ * 
+ * The \a points array is subdivided into two sub-arrays. The first sub-array,
+ * consisting of the first \a num_resp consecutive elements, contains the
+ * points that this process is responsible for propagating during
+ * \ref p8est_transfer_search. The second sub-array, consisting of the
+ * remaining elements, contains the points known to this process that it is
+ * not responsible for propagating.
+ * 
+ * This structure is intended to be initialised on each process, storing a
+ * disjoint subset of the global set of points. Initially, \a num_resp
+ * should be set to the length of \a points so that each each process is
+ * responsible for propagating all of the points it knows. Users initializing
+ * in other ways should be aware that if no process is responsible for
+ * propagating a point then the point will be forgotten after calling
+ * \ref p8est_transfer_search, and similarly that if multiple processes are
+ * responsible for propagating a point then it will be duplicated.
+ * 
+ * Calling \ref p8est_transfer_search performs the transfer of points and
+ * updates \a num_resp, while preserving the property that each point has
+ * exactly one process responsible for propagating it.
+ * 
+ * Users may modify the points array between calls to
+ * \ref p8est_transfer_search. For example, point coordinates could
+ * be modified to represent movement of points as a simulation evolves through
+ * time. Care should be taken when adding or deleting points, and when
+ * modifying the order of \a points, to ensure that \a num_resp is updated
+ * and that the first \a num_resp points are still the points that the
+ * process should propagate. 
+ * 
+ * During the transfer of points in \ref p8est_transfer_search the \a points
+ * array is destroyed and reallocated. Thus users should not maintain pointers
+ * to it or its contents.
+ */
+typedef struct p8est_transfer_search 
```

Comment:
I'd be fine suffixing the struct name with _s to disambiguate for doxygen.

Reply:
I've opted to rename the struct to p4est_points_context. Problem solved -- for now.